docs/common/widget.ldoc

```diff
@@ -39,3 +39,64 @@
 --- Set/get a widget's buttons.
 -- @param _buttons The table of buttons that should bind to the widget.
 -- @function buttons
+
+
+--- When the layout (size) change.
+-- @signal widget::layout_changed
+
+--- When the widget content changed.
+-- @signal widget::redraw_needed
```

Comment:
Here I'd say something about the result of `:draw()`, similar to the previous point...
The difference between the two signals should become clear.